core/config/project_settings.cpp

```diff
@@ -1492,6 +1492,10 @@ ProjectSettings::ProjectSettings() {
 	GLOBAL_DEF("display/window/subwindows/embed_subwindows", true);
 	// Keep the enum values in sync with the `DisplayServer::VSyncMode` enum.
 	custom_prop_info["display/window/vsync/vsync_mode"] = PropertyInfo(Variant::INT, "display/window/vsync/vsync_mode", PROPERTY_HINT_ENUM, "Disabled,Enabled,Adaptive,Mailbox");
+
+	GLOBAL_DEF("display/window/frame_pacing/android/enable_frame_pacing", true);
+	GLOBAL_DEF(PropertyInfo(Variant::INT, "display/window/frame_pacing/android/swappy_mode", PROPERTY_HINT_ENUM, "pipeline_forced_on,auto_fps_pipeline_forced_on,auto_fps_auto_pipeline"), 2);
```

Comment:
In theory for OpenXR the swapchain on Android should never be created so I'm assuming this logic will thus be completely ignored.
But it's probably smart to disable it in this scenario just to be safe.

Reply:
Hm, looking in `rendering_device_driver_vulkan.cpp`, I think some amount of the Swappy-related code would run, even with OpenXR. It's `VulkanHooks` that allows OpenXR to take over the Vulkan setup, but it appears like some of the setup code would still happen.

So, I agree, it would be good to force Swappy to be disabled if OpenXR is being used.

Reply:
Is there an easy/quick way I can check we're running on OpenXR so I can simply skip Swappy code?

Reply:
An indirect way could be checking if `VulkanHooks::get_singleton() != nullptr`? I don't know if this is a safe assumption, but we could assume that if something is taking over creation of the Vulkan device, then we don't want to use Swappy?

There's more direct checks we could do, but it would mean code under `drivers/vulkan` interacting with code from `modules/openxr` which isn't normally allowed. Or, I suppose we could look at project settings, but that isn't ideal either.

Thoughts? cc @m4gr3d @BastiaanOlij

Reply:
I went for your solution `VulkanHooks::get_singleton() != nullptr` because it felt like the cautious thing to do (avoid Swappy) when we're not controlling device creation and possibly swapchain and presentation.

If someone wants to take this more fine grained like adding VulkanHooks::get_singleton()->allow_swappy() or whatever, that can be done in another PR.